js/misc/paygManager.js

```diff
@@ -0,0 +1,344 @@
+    }
+
+    _onProxyConstructed(object, res) {
+        let success = false;
+        try {
+            success = object.init_finish (res);
+        } catch (e) {
+            logError(e, "Error while constructing D-Bus proxy for " + EOS_PAYG_NAME);
+        }
+
+        if (success) {
+            // Don't use the setters here to prevent emitting a -changed signal
+            // on startup, which is useless and confuses the screenshield when
+            // selecting the session mode to construct the right unlock dialog.
+            this._enabled = this._proxy.Enabled;
+            this._expiryTime = this._proxy.ExpiryTime;
+            this._rateLimitEndTime = this._proxy.RateLimitEndTime;
+            this._setCodeFormat(this._proxy.CodeFormat || "^[0-9]{8}$");
+
+            this._propertiesChangedId = this._proxy.connect('g-properties-changed', this._onPropertiesChanged.bind(this));
+            this._codeExpiredId = this._proxy.connectSignal('Expired', this._onCodeExpired.bind(this));
+
+            this._maybeNotifyUser();
+            this._updateExpirationReminders();
+        }
+
+        this._initialized = true;
+        this.emit('initialized');
+    }
+
+    _onPropertiesChanged(proxy, changedProps, invalidatedProps) {
+        let propsDict = changedProps.deep_unpack();
+        if (propsDict.hasOwnProperty('Enabled'))
+            this._setEnabled(this._proxy.Enabled);
+
+        if (propsDict.hasOwnProperty('ExpiryTime'))
+            this._setExpiryTime(this._proxy.ExpiryTime);
+
+        if (propsDict.hasOwnProperty('RateLimitEndTime'))
+            this._setRateLimitEndTime(this._proxy.RateLimitEndTime);
+
+        if (propsDict.hasOwnProperty('CodeFormat'))
+            this._setCodeFormat(this._proxy.CodeFormat, true);
+    }
+
+    _setEnabled(value) {
+        if (this._enabled === value)
+            return;
+
+        this._enabled = value;
+        this.emit('enabled-changed', this._enabled);
+    }
+
+    _setExpiryTime(value) {
+        if (this._expiryTime === value)
+            return;
+
+        this._expiryTime = value;
+        this._updateExpirationReminders();
+
+        this._paygNotifier.clearNotification();
+        this.emit('expiry-time-changed', this._expiryTime);
+    }
+
+    _setRateLimitEndTime(value) {
+        if (this._rateLimitEndTime === value)
+            return;
+
+        this._rateLimitEndTime = value;
+        this.emit('rate-limit-end-time-changed', this._rateLimitEndTime);
+    }
+
+    _setCodeFormat(value, notify=false) {
+        if (this._codeFormat === value)
+            return;
+
+        this._codeFormat = value;
+        try {
+            this._codeFormatRegex = new GLib.Regex(this._codeFormat,
+                                                   GLib.RegexCompileFlags.DOLLAR_ENDONLY,
+                                                   GLib.RegexMatchFlags.PARTIAL);
+        } catch (e) {
+            logError(e, `Error compiling CodeFormat regex "${this._codeFormat}"`);
```

Comment:
This kind of string breaks gettext translations :(

Better not use it.

Reply:
(But will be fixed at a very near future; the patch at https://savannah.gnu.org/bugs/?50920 landed)

Reply:
But this is a log statement - would it be translated?

Reply:
All the translations for **the entire file** will be broken. If there's any chance this file will receive a translatable string, now or in the future, this should be dropped.